openslides_backend/action/actions/topic/json_upload.py

```diff
@@ -0,0 +1,143 @@
+    DONE = "done"
+
+
+@register_action("topic.json_upload")
+class TopicJsonUpload(DuplicateCheckMixin, Action):
+    """
+    Action to allow to upload a json. It is used as first step of an import.
+    """
+
+    model = Topic()
+    schema = DefaultSchema(Topic()).get_default_schema(
+        additional_required_fields={
+            "data": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        **model.get_properties("title", "text"),
+                        **{
+                            prop: agenda_creation_properties[prop]
+                            for prop in (
+                                "agenda_comment",
+                                "agenda_type",
+                                "agenda_duration",
+                            )
+                        },
+                    },
+                    "required": ["title"],
+                    "additionalProperties": False,
+                },
+                "minItems": 1,
+                "uniqueItems": False,
+            },
+            "meeting_id": required_id_schema,
+        }
+    )
+    permission = Permissions.AgendaItem.CAN_MANAGE
+    headers = [
+        {"property": "title", "type": "string"},
+        {"property": "text", "type": "string"},
+        {"property": "agenda_comment", "type": "string"},
+        {"property": "agenda_type", "type": "string"},
+        {"proptery": "agenda_duration", "type": "number"},
+    ]
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        data = instance.pop("data")
+
+        # enrich data with meeting_id
+        for entry in data:
+            entry["meeting_id"] = instance["meeting_id"]
+
+        # validate and check for duplicates
+        self.init_duplicate_set(instance["meeting_id"])
+        self.rows = [self.validate_entry(entry) for entry in data]
+
+        # generate statistics
+        itemCount, itemNew, itemError = 0, 0, 0
+        for entry in self.rows:
+            itemCount += 1
+            if entry["status"] == ImportStatus.NEW:
+                itemNew += 1
+            if entry["status"] == ImportStatus.ERROR:
+                itemError += 1
+        self.statistics = {
+            "total": itemCount,
+            "created": itemNew,
+            "omitted": itemError,
+        }
+
+        # store rows in the action_worker
+        self.new_store_id = self.datastore.reserve_id(collection="action_worker")
+        fqid = fqid_from_collection_and_id("action_worker", self.new_store_id)
+        self.datastore.write_action_worker(
+            WriteRequest(
+                events=[
+                    Event(
+                        type=EventType.Create,
+                        fqid=fqid,
+                        fields={
+                            "id": self.new_store_id,
+                            "result": {"import": "topic", "rows": self.rows},
```

Comment:
set the `timestamp` field of action worker to current timestamp and the status-field to `running

Reply:
Add `created`, `timestamp` and `state` running to the json_upload and `timestamp` and `state` to import_
I rebased the PR.